video/coktel_decoder.cpp

```diff
@@ -1543,6 +1543,93 @@ Graphics::PixelFormat IMDDecoder::getPixelFormat() const {
 	return Graphics::PixelFormat::createFormatCLUT8();
 }
 
+class DPCMStream : public Audio::AudioStream {
+public:
+	DPCMStream(Common::SeekableReadStream *stream, int rate, int channels, bool oldStereo) {
+		_stream = stream;
```

Comment:
Thanks for your review!

`stream` could be asserted, though if it’s null here you’re probably having a bad day already since it means that a critical memory allocation failed. `channels` is never going to be any value other than 1 or 2 since that argument is always hard-coded and the original data format doesn’t support more than 2 channels, so I don’t think there is any practical reason to check that input.

FWIW the only thing that actually changed here was the addition of the `_oldStereo` flag and an extra sanity check on the audio buffer when reading samples since the buffer can be filled with an odd number of samples. All this code just had to be moved up a bit so the class could be used by functions defined earlier in the source.